tracer/test/Datadog.Trace.Tests/SpanTests.cs

```diff
@@ -360,5 +360,15 @@ public void SetTag_Double()
             span.GetTag(stringKey).Should().Be(stringValue);
             span.GetTag(numericKey).Should().BeNull();
         }
+
+        [Fact]
+        public void ServiceOverride_HasBaseService()
```

Comment:
Should we add a few more tests?
- setting the service name to the same value does not add the tag
- never setting the service name does not add the tag
- changing the service name several time keeps only the original tag value